docs/reference/release-notes/v2.29/index.md

```diff
@@ -0,0 +1,162 @@
+- Automatically add this name to your `siteMetadata` and `package.json`
+- Add `-y` flag. This flag bypasses all prompts other than naming your site
+
+The regular `gatsby-cli` received a new command to list out all plugins in your site by running `gatsby plugin ls`.
+
+## Experimental: Parallel data sourcing
+
+In [v2.28](../v2.28/index.md#experimental-parallel-data-sourcing) we gave a sneak peak at a new feature that enables parallel data sourcing. As a recap:
+
+> Plugin APIs in Gatsby run serially. Generally this what we want as most API calls are CPU/IO bound so things are fastest letting each plugin have the full undivided attention of your computer. But source plugins are often _network_ bound as they're hitting remote APIs and waiting for responses. We tried [changing the invocation of `sourceNodes` to parallel](https://github.com/gatsbyjs/gatsby/pull/28214) on a few sites with 4+ source plugins and saw a big speedup on sourcing (40%+) as they were no longer waiting on each other to start their API calls.
+
+You're now able to activate this experiment in the stable release of Gatsby by adding a flag to your `gatsby-config.js`:
+
+```js
+// In your gatsby-config.js
+module.exports = {
+  flags: {
+    PARALLEL_SOURCING: true,
+  },
+  plugins: [], // your plugins stay the same
+}
+```
+
+Please give us your feedback in the [umbrella discussion](https://github.com/gatsbyjs/gatsby/discussions/28336).
+
+## Performance improvements
+
+We were able to ship a bunch of performance improvements both to Gatsby itself and its plugins:
+
+### `gatsby`
+
+- Improve query running performance for sites with large amount of data (up to 10% in our tests). See https://github.com/gatsbyjs/gatsby/pull/28525 for details.
+
+### `gatsby-source-contentful`
+
+- Improve performance of data sourcing for large Contentful spaces. See https://github.com/gatsbyjs/gatsby/pull/28375 for details.
+- Prevent concurrent requests to Contentful's image processing API. See https://github.com/gatsbyjs/gatsby/pull/28438 for details.
+
+## Slugify option for File System Route API
+
+The File System Route API uses [slugify](https://github.com/sindresorhus/slugify) to create slugs for the generated routes. You're now able to pass custom options to that instance, e.g. when you want to change the separator. The full details are listed in the [README](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-page-creator) of `gatsby-plugin-page-creator`.
+
+## gatsby-image codemod
+
+We introduced some API changes for working with images when we published the new `gatsby-plugin-image` in [v2.26](../v2.26/index.md#gatsby-plugin-image010-beta). In order to make it easier to migrate your code to work with the new plugin, we've created a codemod.Follow the migration instructions in the [README](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-image#upgrading-from-the-gatsby-image2) in order to run the codemod against your project.
```

Comment:
Add a missing space (`created a codemod.Follow` to `created a codemod. Follow`):
```suggestion
We introduced some API changes for working with images when we published the new `gatsby-plugin-image` in [v2.26](../v2.26/index.md#gatsby-plugin-image010-beta). In order to make it easier to migrate your code to work with the new plugin, we've created a codemod. Follow the migration instructions in the [README](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-image#upgrading-from-the-gatsby-image2) in order to run the codemod against your project.
```